config/initializers/sidekiq.rb

```diff
@@ -4,6 +4,28 @@
 require 'sidekiq/web'
 require './config/environment'
 
+redis_params = {
+  host: APP_CONFIG.redis_host ||= 'localhost',
+  port: APP_CONFIG.redis_port ||= '6379',
+  password: APP_CONFIG.redis_password ||= nil
+}
+
```

Comment:
Should `redis_db` also be a parameter instead of hard coding `0`?

Reply:
That is the redis database index. I think, for now, I'll leave it unconfigurable since it's not really a descriptive option to pass in. From what I saw in the redis docs, it has to be an integer between 0 and 15.

Probably will make namespace configurable first, but outside the scope of this PR for now

Reply:
My thought was if you’ve chosen to use an external redis instance, you might very well already be using db 0 for something else.

Reply:
That's possible, however even in the production systems I've worked with, I've never seen someone set the database index, always preferring a namespace over DB index. It's too ephemeral of a value. To give an example, most sidekiq and rails examples don't include the database index in the as an option, rather either a `url` configuration option is made available and the user is expected to know the full url (including the DB index) or they don't provide the index at all.

If everyone is OK with changing the options to be `url` instead of `host`, `port`, and `index`, then I'd be ok with that. But, in my experience, db index is rarely given as a specific option in and of itself..... granted making host/port individual options is pretty rare as well.

Reply:
Never seen it?  I've both used and come across it lots, (but doing some more reading, now see that this was [never a good idea](https://redislabs.com/blog/7-redis-worst-practices/). They should have really made this more obvious in the actual documentation ).

FWIW, lots of Puppet users are also Foreman users.  They might already be using redis.  This is where I came across it most recently and for some reason they use DB 6. https://github.com/theforeman/puppet-foreman/blame/e3bf011200d2519895627e29f0cb458dc47338b7/manifests/config.pp#L14

A `redis_url` would work for me too.  I perhaps prefer that over preaching to users how they can and can't use redis.

Reply:
Because someone is bound to stumble across this thread in years to come...

The redis-essentials book might actually be wrong about the feature being deprecated.
It appears it was deprecated and then later [undeprecated](https://stackoverflow.com/questions/16221563/whats-the-point-of-multiple-redis-databases#comment39227306_16243787) again. (Although it's still discouraged.)

Anyway, final solution looks great.